frame/balances/src/lib.rs

```diff
@@ -885,13 +913,14 @@ pub mod pallet {
 			let result = T::AccountStore::try_mutate_exists(who, |maybe_account| {
 				let is_new = maybe_account.is_none();
 				let mut account = maybe_account.take().unwrap_or_default();
-				let did_provide = account.free >= T::ExistentialDeposit::get();
+				let did_provide =
+					account.free >= Self::ed() && frame_system::Pallet::<T>::providers(who) > 0;
```

Comment:
So this edge-case prevails even with >0 ED now?

Reply:
ahh true we can remove that code when the feature is disabled.

Reply:
No idea what you mean. The code was fundamentally correct in the case that ED>0. The problem was that with ED=0 accounts *should* default to a provider ref of 1 since `balance >= ED` is already met. We can't do this since it may be the System pallet which introduces a default balance, not the Balances pallet, so there's no way of incrementing a reference at the System-level during default-account construction.

Reply:
Okay its not the same in case the feature is enabled but ED not set to zero :+1: